packages/pipeline-editor/src/PipelineService.tsx

```diff
@@ -124,4 +129,54 @@ export class PipelineService {
       buttons: [Dialog.okButton()]
     });
   }
+
+  static convertPipeline(pipelineDefinition: any): any {
+    const pipelineJSON = JSON.parse(JSON.stringify(pipelineDefinition));
+
+    if (!Utils.hasPipelineAppdataField(pipelineJSON.pipelines[0], 'version')) {
```

Comment:
I recommend this be method that gets the version from the pipeline.  In this case, that method will return an integer of 0.

I would then move the contents of this block to a method named something like `convertV0ToV1()`.  This way, when version 2 comes along, this code would resemble something like (using Python):

```python
version = getPipelineVersion(pipelineJSON.pipelines[0])
if version < 1:
    pipelineJSON = convertV0toV1(pipelineJSON)
if version < 2:
    pipelineJSON = convertV1toV2(pipelineJSON)   
if version < 3:
    pipelineJSON = convertV2toV3(pipelineJSON)   
```
and each of the "convert" methods only needs to address the specific changes for that particular transition.

Reply:
Should the pipeline file get saved if conversions are performed?

Reply:
After hearing the discussion about updates and warning for updates I'm okay with whatever approach.  I guess I'd prefer to not get prompted, but since this is an automated update, it makes sense to prompt.  If we're going to prompt at all, I think it should be to confirm/approve the update vs a warning/info dialog that their pipeline was updated.

Reply:
for the save, i would prefer the user to save it, which gives a way for a real person validation.